woocommerce/woocommerce-android
# Issue/5876 dismiss all notes

Closes: #5876 


### Description
This PR adds a button on the toolbar to enable dismissing all Inbox notes at once. 

How the API endpoint: `/wc-analytics/admin/notes/delete/all` works is kind of "special" to say the least 😅 . Basically, you can't just make a request to that endpoint and expect all notes to be deleted. You have to delete notes in a paginated way. This pagination is limited to delete notes at maximum rate of 25 at the time. So if you have more than 25 notes in your Inbox, and user hits the "Dismiss All" notes button we will have to make as many requests as necessary to remove all the notes in blocks of 25 or less notes. That is not handled in this changes though it is handled in alrady merged changes in [FluxC](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2355). 

### Testing instructions
- Go to More Menu -> Inbox notes. 
- Click overflow menu -> Dismiss all. 
- You should see an empty Inbox
- Navigate back 
- Enter again in the Inbox and wait until notes are fetched from the API. The screen should still be empty. 
- You can also check that the inbox in `wc-core` is empty

### Images/gif


https://user-images.githubusercontent.com/2663464/163374226-30330c43-8340-4bcf-a9db-80d09714c31b.mp4
